ipsource.go

```diff
@@ -371,21 +381,27 @@ func (s *Static) Provision(ctx caddy.Context) error {
 
 func (s *Static) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	d.Next() // skip directive name
-
 	for d.NextArg() {
-		raw_ip := d.Val()
-		ip := net.ParseIP(raw_ip)
-
-		if ip == nil {
-			return d.Errf("Invalid IP address: %v", raw_ip)
+		raw := d.Val()
+		addr, err := netip.ParseAddr(raw)
+		if err != nil {
+			return d.Errf("Invalid IP address '%s': %w", raw, err)
 		}
-
-		s.IPs = append(s.IPs, ip)
+		s.IPs = append(s.IPs, addr)
 	}
-
 	return nil
 }
 
+// IPNetToNetIPPrefix converts a [net.IPNet] to a [netip.Prefix].
+func IPNetToNetIPPrefix(ipNet *net.IPNet) netip.Prefix {
```

Comment:
Do you think this has to be exported?

Reply:
No, I was so focused on writing the code it never occurred to me that this is a public package haha.